src/background.js

```diff
@@ -254,44 +269,11 @@ function shouldContainInto (url, tab) {
   return false;
 }
 
-async function maybeReopenAlreadyOpenTabs () {
-  const maybeReopenTab = async tab => {
-    const macAssigned = await getMACAssignment(tab.url);
-    if (macAssigned) {
-      // We don't reopen MAC assigned urls
-      return;
-    }
-    const cookieStoreId = shouldContainInto(tab.url, tab);
-    if (!cookieStoreId) {
-      // Tab doesn't need to be contained
-      return;
-    }
-    reopenTab({
-      url: tab.url,
-      tab,
-      cookieStoreId
-    });
-  };
-
-  const tabsOnUpdated = (tabId, changeInfo, tab) => {
-    if (changeInfo.url && tabsWaitingToLoad[tabId]) {
-      // Tab we're waiting for switched it's url, maybe we reopen
-      delete tabsWaitingToLoad[tabId];
-      maybeReopenTab(tab);
-    }
-    if (tab.status === "complete" && tabsWaitingToLoad[tabId]) {
-      // Tab we're waiting for completed loading
-      delete tabsWaitingToLoad[tabId];
-    }
-    if (!Object.keys(tabsWaitingToLoad).length) {
-      // We're done waiting for tabs to load, remove event listener
-      browser.tabs.onUpdated.removeListener(tabsOnUpdated);
-    }
-  };
-
+async function handleAlreadyOpenTabs () {
   // Query for already open Tabs
   const tabs = await browser.tabs.query({});
   tabs.map(async tab => {
+    updateBrowserActionIcon(tab);
```

Comment:
Does this need to be for all tabs here or just the current selected?

Reply:
You're right. Since the onUpdate'd listener also runs when switching tabs (because `active` changes), we can call `updateBrowserActionIcon` only once for the currently active tab.